### ISSUE TITLE
Thumbnails are not showing

### ISSUE THREAD

**Issue by georgi-karadzhov** [saschanaz/jxl-winthumb#41]
After registering the thumbnails change from this:

![image](https://github.com/saschanaz/jxl-winthumb/assets/171819041/a10514c1-e41f-424c-b2b2-04f1e3064a52)

to this:

![image](https://github.com/saschanaz/jxl-winthumb/assets/171819041/38a409cf-b76c-4c6f-ab38-424d2522655e)

**Comment by saschanaz** [comment]
Hello! Are all files affected or only a certain file is?

**Comment by georgi-karadzhov** [comment]
> Hello! Are all files affected or only a certain file is?


Hi, all files are affected.

**Comment by saschanaz** [comment]
That sounds bad, have you tried restarting your machine?

**Comment by saschanaz** [comment]
And if you double click the file, does it open the Windows Photo Gallery and show the image?

**Comment by georgi-karadzhov** [comment]
After restart, it is the same.
Yes, if I double click it, the file is opened without problems.

**Comment by saschanaz** [comment]
That means the installation is done properly. If you refresh in the file explorer, does anything happen?

**Comment by georgi-karadzhov** [comment]
No, nothing happens.

**Comment by saschanaz** [comment]
Hmm. What OS architecture do you use and which dll did you install?

**Comment by saschanaz** [comment]
And have you tried reinstalling (either after uninstall or not)?

**Comment by georgi-karadzhov** [comment]
I am using Windows 10 22H2 (Build 19042.4412) and installed the _jxl_winthumb.dll_.
I tried reinstalling but it didn't fix it.

**Comment by keftie22** [comment]
Having the same problem on Windows 11 23H2 22631.3737. Icons are all white blanks yet they open in the old windows photo viewer program.

![image](https://github.com/saschanaz/jxl-winthumb/assets/101351551/75e55440-6b18-4b71-ad8c-d92a1b1bf2c5)

**Comment by keftie22** [comment]
Checked again a week later and suddenly it's working now. Even a restart before didn't fix it but somehow it's been fixed for me. I am having a separate issue with it though that I will open another issue for.

**Comment by Guilherme-Weber** [comment]
hey, i was having the same issue, i deleted the thumbnail cache on windows using the "disk cleanup" tool and then restarted explorer, and it was fixed!!!

**Comment by haico1992** [comment]
Hvaing the same problem, but clean Thumbnail cache doesn't work for me.
Windows 11 Pro , 23H2
![image](https://github.com/user-attachments/assets/a0a572d0-9896-4c07-8cbd-e4177f31c2ab)

**Comment by UpscaleAnon** [comment]
Just tried jxl-winthumb and JXL files do not display a thumbnail, they just show an icon for the default app you use to view them. Instructions say to use jxl_winthumb.dll, but latest version doesn't have that jxl_winthumb_x86_64.dll and such and I assume that's the version to use for 64 bit systems, but it doesn't work.
I've also tried previous version 0.8.1 which actually has a "jxl_winthumb.dll" but that doesn't work either.
I've restarted explorer, PC, updated Windows, updated GPU drivers and cleared thumbnail cache but no luck there.
I did not however try restarting PC after the 0.9.0 version with the jxl_winthumb_x86_64.dll, but I can't imagine 1 version ahead working when previous didn't.
I'm using Windows 10 Home 22H2.
And according to CMD using tasklist, it shows explorer.exe using jxl_winthumb.dll
However, as I'm writing this, I just went ahead and did "regsvr32 jxl_winthumb.dll" for a second time, and now everything magically works, which makes no sense.
A bit of a painful experience just to get some damn thumbnails lol. I wish JXL behaved like every other image format and just displayed normally without requiring any of this.

**Comment by saschanaz** [comment]
It's weird, for me it always worked at one shot, not sure what's happening. Maybe the installation hook can somehow force removing thumbnail cache?

**Comment by AndroidDeveloperLB** [comment]
Any news about this? For some reason I have some files that are shown fine, and some that aren't.
See attached.

[test.zip](https://github.com/user-attachments/files/17188650/test.zip)

**Comment by saschanaz** [comment]
If only some files are affected it might be possible that it's a decoder bug 👀 I'll check...

**Comment by AndroidDeveloperLB** [comment]
@saschanaz If it's important, I've saved the file using "Paint.Net" app, but same occurs on these websites:
https://ezgif.com/webp-to-jxl
https://www.vertopal.com/

I also wonder if it's related to the fact that on Chrome&Firefox, they fail to show some (all except the first) JPEG-XL files even after installing addons to support it, as I've tested here:
https://jpegxl.info/resources/jpeg-xl-test-page

**Comment by saschanaz** [comment]
I'm still outside, can djxl decode those files?

**Comment by AndroidDeveloperLB** [comment]
@saschanaz I don't know about specific libraries, but I know Paint.Net app can open the files.

**Comment by TeKett** [comment]
Same problem, win10 22H2. 

I think i know the issue. The new photo viewer refuses the recognize the file, windows does not allow photo viewer to be select with "open with" (asinine windows decisions). Can someone figure out how to forcefully set the .jxl file extension to be associated with the new win10/win11 photo app? I have tried for an hour now and cant figure it out. You cant run the exe directly and "open with" can't open the "whatever they are" in the virtual applications folder (the ones microsoft wants use to use rather then the exe's). 

Does Linux support .JXL? Since that would be another reason to swap after microsoft drops windows10 support.

**Comment by saschanaz** [comment]
Photos app does not support Windows Imaging Component (that this library depends on) and is completely independent from the thumbnails, which is processed by File Explorer.

Still mysterious why some people suffer from this problem.

**Comment by supermimai** [comment]
@saschanaz  do you have any application to suggest using wic to display pictures ? at least to validate if the wic works or not ( also can you confirm the status of exif / iptc support thru wic )

**Comment by saschanaz** [comment]
If you can double click the image and that properly opens the old Windows Photo Viewer, then that means WIC component is properly up and running.

**Comment by saschanaz** [comment]
> exif / iptc support thru wic 

That's #26 and not WIC related. Help appreciated.

**Comment by TeKett** [comment]
> Photos app does not support Windows Imaging Component (that this library depends on) and is completely independent from the thumbnails, which is processed by File Explorer.
> 
> Still mysterious why some people suffer from this problem.

From what i can see, whenever i open a folder with images lacking a thumbnail dllhost is run together with some connection to the photos app.

When i accidentally clicked a bit to much, the thumbnails and previews froze, which is more common then it sounds, not an issue with this project. The problem is that  dllhost is waiting on 'photos' that is lacking a thread, during this time no thumbnails or previews are generated, if you click on a different file preview does not change. Terminating that specific dllhost fixes the problem. 

So it does at least in some way interfere or cross paths, if not is part of the generation.

**Comment by ValiumBear** [comment]
It's still not working for me to this day also, it's very unfortunate there's still no fix for this.
Is it possible you missed something?

For example I can see it's listed as Property Handler and no file extensions specified, whereas Icaros are using 3 separate - Property Handler, Thumbnail, Thumbnail Handler (and the last 2 for thumbnail also specify to which file extensions they apply)

![Image](https://github.com/user-attachments/assets/52d4464e-89b4-425f-8c01-2da2ef5e6311)
![Image](https://github.com/user-attachments/assets/af2b7023-2b93-4f5f-b104-23c56dd9d853)
![Image](https://github.com/user-attachments/assets/87b0f69b-84a5-45a3-b4f7-e3c6c0ef77fd)

Edit: Also found Microsoft's own already applying to .jxl and other types (but no thumbnails either).
Could it be a conflict?

![Image](https://github.com/user-attachments/assets/bf572348-e911-477d-978b-3efed99c274c)

Edit2: Found a fix. Uninstalled HEIF image extension (didn't work on it's own), WebP image extension, cleared thumbnail cache and killed explorer, then it worked.

So it seems the WebP extension was causing conflicts. Can others test?
